controlplane/kubeadm/internal/workload_cluster.go

```diff
@@ -503,6 +509,19 @@ func checkStaticPodReadyCondition(pod *corev1.Pod) error {
 	return nil
 }
 
+func checkNodeNoExecuteCondition(node *corev1.Node) error {
+	found := false
+	for _, taint := range node.Spec.Taints {
+		if taint.Key == corev1.TaintNodeUnreachable && taint.Effect == corev1.TaintEffectNoExecute {
+			found = true
+		}
+	}
+	if found {
+		return errors.Errorf("node has NoExecute taint: %v", node.Name)
+	}
+	return nil
```

Comment:
I think you can simplify this to `return nil` when it finds the match, and returns an error if it doesn't. (i.e. remove the `found` logic)

Reply:
edited the above as i misread the original logic